drivers/can/can_numaker.c

```diff
@@ -0,0 +1,316 @@
+}
+
+static int can_numaker_write_reg(const struct device *dev, uint16_t reg, uint32_t val)
+{
+	const struct can_mcan_config *mcan_cfg = dev->config;
+	const struct can_numaker_config *numaker_cfg = mcan_cfg->custom;
+
+	return can_mcan_sys_write_reg(numaker_cfg->canfd_base, reg, val);
+}
+
+static int can_numaker_read_mram(const struct device *dev, uint16_t offset, void *dst, size_t len)
+{
+	const struct can_mcan_config *mcan_cfg = dev->config;
+	const struct can_numaker_config *numaker_cfg = mcan_cfg->custom;
+
+	return can_mcan_sys_read_mram(numaker_cfg->mram, offset, dst, len);
+}
+
+static int can_numaker_write_mram(const struct device *dev, uint16_t offset, const void *src,
+				size_t len)
+{
+	const struct can_mcan_config *mcan_cfg = dev->config;
+	const struct can_numaker_config *numaker_cfg = mcan_cfg->custom;
+
+	return can_mcan_sys_write_mram(numaker_cfg->mram, offset, src, len);
+}
+
+static int can_numaker_clear_mram(const struct device *dev, uint16_t offset, size_t len)
+{
+	const struct can_mcan_config *mcan_cfg = dev->config;
+	const struct can_numaker_config *numaker_cfg = mcan_cfg->custom;
+
+	return can_mcan_sys_clear_mram(numaker_cfg->mram, offset, len);
+}
+
+static const struct can_mcan_ops can_numaker_ops = {
+	.read_reg = can_numaker_read_reg,
+	.write_reg = can_numaker_write_reg,
+	.read_mram = can_numaker_read_mram,
+	.write_mram = can_numaker_write_mram,
+	.clear_mram = can_numaker_clear_mram,
+};
+
+#define NUMAKER_CLKCTRL_DEV_INIT(inst)          \
+	.clk_dev    = DEVICE_DT_GET(DT_PARENT(DT_INST_CLOCKS_CTLR(inst))),
+
+#define NUMAKER_PINCTRL_DEFINE(inst)            \
+	PINCTRL_DT_INST_DEFINE(inst);
+#define NUMAKER_PINCTRL_INIT(inst)              \
+	.pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(inst),
+
+#define CAN_NUMAKER_INIT(inst)                  \
+	NUMAKER_PINCTRL_DEFINE(inst);               \
+	CAN_MCAN_DT_INST_CALLBACKS_DEFINE(inst, can_numaker_cbs_##inst);	    \
+												\
+	static void can_numaker_irq_config_func_##inst(const struct device *dev);         \
+												\
+	static const struct can_numaker_config can_numaker_config_##inst = {              \
+		.canfd_base = DT_INST_REG_ADDR_BY_NAME(inst, m_can),                \
+		.mrba = CAN_MCAN_DT_INST_MRBA(inst),			    \
+		.mram = CAN_MCAN_DT_INST_MRAM_ADDR(inst),			    \
+		.reset = RESET_DT_SPEC_INST_GET(inst),                                             \
+		.clk_modidx = DT_INST_CLOCKS_CELL(inst, clock_module_index),    \
+		.clk_src    = DT_INST_CLOCKS_CELL(inst, clock_source),          \
+		.clk_div    = DT_INST_CLOCKS_CELL(inst, clock_divider),         \
+		NUMAKER_CLKCTRL_DEV_INIT(inst)                                  \
+		.irq_config_func    = can_numaker_irq_config_func_##inst,     \
+		NUMAKER_PINCTRL_INIT(inst)                                      \
+	};                                          \
+												\
+	static const struct can_mcan_config can_mcan_config_##inst =      \
```

Comment:
the `\` here are all over the place with pretty much any tabstop value (and the right one is 8 anyway :-), could you align them?

Reply:
of course, I aligned them.